src/nvim/api/buffer.c

```diff
@@ -1420,10 +1420,10 @@ Integer nvim_buf_add_highlight(Buffer buffer,
     end_line++;
   }
 
-  ns_id = extmark_add_decoration(buf, ns_id, hlg_id,
-                                 (int)line, (colnr_T)col_start,
-                                 end_line, (colnr_T)col_end,
-                                 VIRTTEXT_EMPTY);
+  extmark_add_decoration(buf, ns_id, hlg_id,
```

Comment:
ref: https://github.com/QnJ1c2kNCg/neovim/commit/ca1a00edd6d6345b848a28d077d6a192528f811e#diff-9c343d9e2b2c6c1a75a00e9e95432767R1423-R1427

Shouldn't this change ns_id to src_id instead of removing ns_id?

Reply:
good question. We have test coverage for this, but maybe there's an edge case... @bfredl should `nvim_buf_add_highlight` return the result of `extmark_add_decoration` ?

Reply:
`src_id` has already been calculated. This PR is correct, just delete the assigment here.